backend/test/v2/integration/cache_test.go

```diff
@@ -65,21 +65,26 @@ func (s *CacheTestSuite) SetupTest() {
 	var newPipelineClient func() (*apiServer.PipelineClient, error)
 	var newRunClient func() (*apiServer.RunClient, error)
 	var newRecurringRunClient func() (*apiServer.RecurringRunClient, error)
-	//todo: this differs from the gpt impl
 	var tlsCfg *tls.Config
+	caCertPool, err := x509.SystemCertPool()
```

Comment:
Could you put this code under `if *tlsEnabled`? Repeat this for all other instances please.